### PR TITLE
Md 157 create logic to make the qrcode

### PR DESCRIPTION
**Pull request by ebertonArouxa**

### CODE REVIEW

File: src/components/Qrcode/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import styles from './styles.module.css';
+
+export default function Qrcode() {
+  const { user } = useUser();
+
+  const fullURL = window.location.href;
+  const urlObj = new URL(fullURL);
+  const pathname = urlObj.origin;
+
+  const qrcodeProductsUrl = `${pathname}/produtos/${user?.publicMetadata.slug}/${user?.id}`;
```
Comment (EricksonSiqueira) [inline src/components/Qrcode/index.tsx:14]:
```suggestion
  const qrCodeProductsUrl = `${pathname}/produtos/${user?.publicMetadata.slug}/${user?.id}`;
```

File: src/components/Qrcode/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  const handleSaveQrcode = () => {
```
Comment (EricksonSiqueira) [inline src/components/Qrcode/index.tsx:16]:
```suggestion
  const handleSaveQrCode = () => {
```

File: src/components/Qrcode/index.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    window.print();
+  };
+
+  return (
+    <>
+      <div className={styles.qrcodeWrapper}>
+        <QRCode value={qrcodeProductsUrl} />
+        <button
+          type="button"
+          className={styles.qrcodeButton}
```
Comment (EricksonSiqueira) [inline src/components/Qrcode/index.tsx:26]:
```suggestion
      <div className={styles.qrCodeWrapper}>
        <QRCode value={qrcodeProductsUrl} />
        <button
          type="button"
          className={styles.qrCodeButton}
```

File: src/components/Qrcode/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import QRCode from 'react-qr-code';
+
+import { useUser } from '@clerk/nextjs';
+
+import styles from './styles.module.css';
+
+export default function Qrcode() {
```
Comment (EricksonSiqueira) [inline src/components/Qrcode/index.tsx:7]:
```suggestion
export default function QrCode() {
```

File: src/components/Qrcode/styles.module.css [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.qrcodeWrapper {
+  @apply flex flex-col w-full h-4/5 justify-center items-center md:h-80 md:pt-3;
+}
+
+.qrcodeButton {
```
Comment (EricksonSiqueira) [inline src/components/Qrcode/styles.module.css:5]:
```suggestion
.qrCodeWrapper {
  @apply flex flex-col w-full h-4/5 justify-center items-center md:h-80 md:pt-3;
}

.qrCodeButton {
```

**Review by EricksonSiqueira — COMMENTED** [review]
This is just a visual opnion, i think that the button would be prettier if it was the same size as the QR Code

image

File: src/components/Qrcode/index.tsx (hunk 1 above)
Comment (ebertonArouxa) [reply src/components/Qrcode/index.tsx]: nice, i changed

File: src/components/Qrcode/index.tsx (hunk 2 above)
Comment (ebertonArouxa) [reply src/components/Qrcode/index.tsx]: done

File: src/components/Qrcode/index.tsx (hunk 3 above)
Comment (ebertonArouxa) [reply src/components/Qrcode/index.tsx]: done

File: src/components/Qrcode/index.tsx (hunk 4 above)
Comment (ebertonArouxa) [reply src/components/Qrcode/index.tsx]: done

File: src/components/Qrcode/styles.module.css (hunk 5 above)
Comment (ebertonArouxa) [reply src/components/Qrcode/styles.module.css]: done

**Review by claudiocassimiro — APPROVED** [review]

**Review by EricksonSiqueira — APPROVED** [review]

**Pull request merged**